File: codewar/V1/Day-171/uniqueInOrder.js
// https://www.codewars.com/kata/54e6533c92449cc251001667/train/javascript

/**
 *
 * @param {String} iterable
 * @returns
 */
const uniqueInOrder = (iterable) => {
  return [...iterable].filter((item, i, arr) => {
    return arr[i] !== arr[i + 1];
  });
};

// const a = "AAAABBBCCDAABBB";
// console.log(uniqueInOrder(a));

/**
 *
 * @param {String} iter
 */
const test = (iter) => {
  let res = [];
  for (let i = 0; i < iter.length; ++i) {
    if (res.length === 0 || iter[i] !== res[-1]) {
      res.push(iter[i]);
    }
  }
  return res;
};

const a = "AAAABBBCCDAABBB";
// console.log(test(a));

File: codewar/V1/Day-171/uniqueInOrder.py
# https://www.codewars.com/kata/54e6533c92449cc251001667/train/python
# i coul
def unique_in_order(iter: str):
    res = []
    list_iter = [char for char in iter]
    for index, value in enumerate(list_iter):
        if value != list_iter[index + 1]:
            res.append(value)

    return res


a = "AAAABBBCCDAABBB"
print(a[-1])

# print(unique_in_order(a))
# i don't know this SHIT


def unique_in_order2(iter: str):
    pass
